### PR TITLE
Migrate existing code to `Legacy` subfolders

### PR DESCRIPTION
**Pull request by keefertaylor**
rippled is going to implement protocol buffer support natively, which is going to use a set of well reviewed protocol buffers. The implementation in rippled is being done here: https://github.com/ripple/rippled/pull/3159.

In order to do the upgrade in place, we will swap out the base decorator with two implementations: 
- `LegacyDefaultXpringClient`: Uses the old protocol buffers
- `DefaultXpringClient`: Uses the new protocol buffers

Once we have a working implementation, the constructor on `XpringClient` will be changed to take a new boolean parameter, `useLegacy`, which will control with base implementation is used. 

I propose doing this upgrade in a few stages:
- [x] [Generate new versions of the protocol buffers](https://github.com/xpring-eng/Xpring-JS/pull/108)
- [x] Migrate existing code to a `legacy` subdirectory and namespace (This PR)
- [ ] Create code that works with the new protocol buffers from the rippled team
- [ ] Add a boolean parameter to switch between implementations

### CODE REVIEW

**Comment by hbergren** [comment]
@keefertaylor , can you rebase this on master so only the relevant commits are included? It looks like it has the same commits that have already been merged into master by https://github.com/xpring-eng/Xpring-JS/pull/108.

**Comment by keefertaylor** [comment]
@hbergren rebased! sorry about that.

File: src/legacy/legacy-grpc-network-client.ts [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+import { Fee } from "../../generated/legacy/fee_pb";
+import { GetFeeRequest } from "../../generated/legacy/get_fee_request_pb";
+import { GetAccountInfoRequest } from "../../generated/legacy/get_account_info_request_pb";
+import { SubmitSignedTransactionRequest } from "../../generated/legacy/submit_signed_transaction_request_pb";
+import { SubmitSignedTransactionResponse } from "../../generated/legacy/submit_signed_transaction_response_pb";
+import { GetLatestValidatedLedgerSequenceRequest } from "../../generated/legacy/get_latest_validated_ledger_sequence_request_pb";
+import { LedgerSequence } from "../../generated/legacy/ledger_sequence_pb";
+import { GetTransactionStatusRequest } from "../../generated/legacy/get_transaction_status_request_pb";
+import { TransactionStatus } from "../../generated/legacy/transaction_status_pb";
+import { LegacyNetworkClient } from "./legacy-network-client";
```
Comment (hbergren) [inline src/legacy/legacy-grpc-network-client.ts]: Any specific reason you changed the import from `* as Networking` to `{ LegacyNetworkClient }`?

File: test/mocha.opts [hunk 2]
```diff
@@ -1,4 +1,4 @@
 --require ./node_modules/ts-node/register
 --require ./node_modules/source-map-support/register
 --recursive
-test/*.ts
\ No newline at end of file
+test/**/*.ts
```
Comment (hbergren) [inline test/mocha.opts]: nit: newline.

If you're using VS Code, you can punch these into your user settings:
```
    "files.insertFinalNewline": true,
    "files.trimFinalNewlines": true,
    "files.trimTrailingWhitespace": true,
```

File: src/legacy/legacy-grpc-network-client.ts (hunk 1 above)
Comment (keefertaylor) [reply src/legacy/legacy-grpc-network-client.ts]: No. I think this was actually just a bad VSCode autocomplete at some point. Any reason you'd like me to namespace this object?

File: test/mocha.opts (hunk 2 above)
Comment (keefertaylor) [reply test/mocha.opts]: https://github.com/xpring-eng/Xpring-JS/pull/112

Comment (keefertaylor) [reply test/mocha.opts]: Added newline. Sent PR to put VSCode settings into project repo.

File: src/legacy/legacy-grpc-network-client.ts (hunk 1 above)
Comment (hbergren) [reply src/legacy/legacy-grpc-network-client.ts]: I can't think of any, it was just a change that stood out to me as different from the other `import` changes.

**Pull request merged**